### PR TITLE
Add Benchmark Task resource

### PR DESCRIPTION
**Pull request by nkraemer-sysdig**
We would like to add the ability to https://github.com/sysdiglabs/terraform-aws-secure-for-cloud to automatically create a benchmark task when using the cloud-bench module. 

This PR adds a Secure Benchmark Task resource.

Tested using a local build and https://github.com/sysdiglabs/terraform-aws-secure-for-cloud/pull/17

### CODE REVIEW

File: website/docs/r/sysdig_secure_cloud_account.md [hunk 1]
```diff
@@ -35,7 +35,7 @@ resource "sysdig_secure_cloud_account" "sample" {
 
 ## Import
 
-Secure Teams can be imported using the `account_id`, e.g.
+Secure Cloud Accounts can be imported using the `account_id`, e.g.
```
Comment (nkraemer-sysdig) [inline website/docs/r/sysdig_secure_cloud_account.md:38]: Missed this in a previous PR

File: sysdig/resource_sysdig_secure_benchmark_task_test.go [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+}
+
+func secureBenchmarkTaskWithName(name string) string {
+	return fmt.Sprintf(`
+resource "sysdig_secure_benchmark_task" "sample" {
+  name     = "%s"
+  schedule = "0 6 * * *"
+  schema   = "aws_foundations_bench-1.3.0"
+  scope    = "aws.accountId = \"123456789012\" and aws.region = \"us-west-2\""
+  enabled  = "true"
```
Comment (nestorsalceda) [inline sysdig/resource_sysdig_secure_benchmark_task_test.go:48]: Let's use a true value here, as a boolean, without the quotes please :)

File: sysdig/resource_sysdig_secure_benchmark_task_test.go [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+	})
+}
+
+func secureBenchmarkTaskWithName(name string) string {
+	return fmt.Sprintf(`
+resource "sysdig_secure_benchmark_task" "sample" {
+  name     = "%s"
+  schedule = "0 6 * * *"
+  schema   = "aws_foundations_bench-1.3.0"
+  scope    = "aws.accountId = \"123456789012\" and aws.region = \"us-west-2\""
```
Comment (nestorsalceda) [inline sysdig/resource_sysdig_secure_benchmark_task_test.go:47]: Would be possible to test the multirregion feature?

File: sysdig/resource_sysdig_secure_benchmark_task_test.go [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+		},
+	})
+}
+
+func secureBenchmarkTaskWithName(name string) string {
+	return fmt.Sprintf(`
+resource "sysdig_secure_benchmark_task" "sample" {
+  name     = "%s"
+  schedule = "0 6 * * *"
+  schema   = "aws_foundations_bench-1.3.0"
```
Comment (nestorsalceda) [inline sysdig/resource_sysdig_secure_benchmark_task_test.go:46]: Would be possible to have a list of the possible schemas, just to be able to detect errors before hitting the API?

File: sysdig/resource_sysdig_secure_benchmark_task_test.go [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+			},
+		},
+	})
+}
+
+func secureBenchmarkTaskWithName(name string) string {
+	return fmt.Sprintf(`
+resource "sysdig_secure_benchmark_task" "sample" {
+  name     = "%s"
+  schedule = "0 6 * * *"
```
Comment (nestorsalceda) [inline sysdig/resource_sysdig_secure_benchmark_task_test.go:45]: I think the BE will handle the "* * * * *" running each minute, is this true?

Running benchmarks every minute may be a bit expensive for the user, and if a lot of uses run this in a minutely way can be a bit resource exhausting.

**Review by nestorsalceda — CHANGES_REQUESTED** [review]

File: sysdig/resource_sysdig_secure_benchmark_task_test.go (hunk 5 above)
Comment (nkraemer-sysdig) [reply sysdig/resource_sysdig_secure_benchmark_task_test.go:45]: This is restricted in the backend (to a maximum of once per day), but the code is relatively complex. I think it makes sense to leave the validation in the backend, and let the provider fail if the schedule is too frequent. 

The once per day maximum schedule is noted in the documentation as well.

File: sysdig/resource_sysdig_secure_benchmark_task_test.go (hunk 4 above)
Comment (nkraemer-sysdig) [reply sysdig/resource_sysdig_secure_benchmark_task_test.go:49]: https://github.com/sysdiglabs/terraform-provider-sysdig/pull/117/commits/ae3b41f02aa99a9df684b0464adf0f73ce12aa26

File: sysdig/resource_sysdig_secure_benchmark_task_test.go (hunk 3 above)
Comment (nkraemer-sysdig) [reply sysdig/resource_sysdig_secure_benchmark_task_test.go:50]: https://github.com/sysdiglabs/terraform-provider-sysdig/pull/117/commits/ae3b41f02aa99a9df684b0464adf0f73ce12aa26

File: sysdig/resource_sysdig_secure_benchmark_task_test.go (hunk 2 above)
Comment (nkraemer-sysdig) [reply sysdig/resource_sysdig_secure_benchmark_task_test.go]: https://github.com/sysdiglabs/terraform-provider-sysdig/pull/117/commits/ae3b41f02aa99a9df684b0464adf0f73ce12aa26

**Comment by nkraemer-sysdig** [comment]
Thanks @wideawakening for catching: https://github.com/sysdiglabs/terraform-provider-sysdig/pull/117/commits/d36477176510acc7fe044a0c6fcced5236127188 

Good _eye_ for detail

File: sysdig/resource_sysdig_secure_benchmark_task_test.go (hunk 5 above)
Comment (nestorsalceda) [reply sysdig/resource_sysdig_secure_benchmark_task_test.go:48]: Makes sense, thank you @nkraemer-sysdig :)

**Pull request merged**